docs/source/Plugin/P004_DS18b20.rst

```diff
@@ -93,10 +93,15 @@ Sensor
 GPIO <--> 1-wire
 """"""""""""""""
 
-Pin used for communicating with the Dallas sensor(s).
+Pin used for communicating with the Dallas sensor(s). This is the single wire setup so you can leave the TX GPIO empty or set it equal to the RX GPIO.
 
 Please note the pin must be pulled up by a resistor, so only use pins that can be in a pulled up state during boot (thus NOT use GPIO 15 on ESP8266).
 
+GPIO <--> Shelly Temperature add-on
+"""""""""""""""""""""""""""""""""""
+
+The Shelly temperature add-on require a two wire setup. Set TX to GPIO 3 and RX to GPIO 0.

```

Comment:
One question about this.
Does that cause any conflict with Serial0?

Reply:
I don't know. What can i do to check?

Reply:
Does it still work when "Use Serial" is enabled in tools -> Advanced?
What happens if you have the Dallas plugin still enabled and trying to connect via serial?

Reply:
> What happens if you have the Dallas plugin still enabled and trying to connect via serial?

I don't think that matters muchm because there is no sensible way you want to have both enabled at the same time on a Shelly.

> Does it still work when "Use Serial" is enabled in tools -> Advanced?

Yes it does.

Reply:
Well if "Use Serial" is enabled, any data received on that port, which may somehow be interpreted as data, is handed over to the command interpreter.
So it may end up giving garbage in the logs, or at least cause extra load for the parsing.
Also I don't know what effect sending the logs to serial may have on the operations of the Dallas task.

Not saying it is useful to have them active at the same time, but at least there should be a warning in the documentation about disabling serial when using those pins for other tasks.

Reply:
> but at least there should be a warning in the documentation about disabling serial when using those pins for other tasks.

Isn't that the reason that those pins have a warning sign attached in the first place. I mean the warning should be there in general, not specified to this plugin. `When using a warning sign GPIO, make sure you disable the accompanied feature attached to it`

Reply:
Yep the warning sign is there if the pins are also actively used via "Use Serial".
But maybe some mentioning in the documentation that GPIO-3 might be used by the serial port can give people a hint where to look.

The warning sign is often just ignored by users. So there has to be some kind of trade-off between "explaining it all" and "being flooded by reported issues or forum questions".

Reply:
> But maybe some mentioning in the documentation that GPIO-3 might be used by the serial port can give people a hint where to look.

On the `Shelly` section of the `004` page?

Reply:
Yep. For others it doesn't really apply.
But on the Shelly you mention very specific pins to use, so users are bound to run into issues as the "Use Serial" is enabled by default.